internal/controller/standalone_pgadmin/users_test.go

```diff
@@ -0,0 +1,599 @@
+		pgadmin.Status.ImageSHA = "fakeSHA"
+		pod := pod.DeepCopy()
+
+		pod.DeletionTimestamp = nil
+		pod.Status.ContainerStatuses =
+			[]corev1.ContainerStatus{{Name: naming.ContainerPGAdmin}}
+		pod.Status.ContainerStatuses[0].State.Running =
+			new(corev1.ContainerStateRunning)
+		pod.Status.ContainerStatuses[0].ImageID = "newFakeSHA"
+
+		r := new(PGAdminReconciler)
+		r.Client = fake.NewClientBuilder().WithObjects(pod).Build()
+
+		calls := 0
+		r.PodExec = func(
+			namespace, pod, container string,
+			stdin io.Reader, stdout, stderr io.Writer, command ...string,
+		) error {
+			calls++
+
+			// Simulate a v7 version of pgAdmin by setting stdout to "7" for
+			// podexec call in reconcilePGAdminMajorVersion
+			stdout.Write([]byte("7"))
+			return nil
+		}
+
+		assert.NilError(t, r.reconcilePGAdminUsers(ctx, pgadmin))
+		assert.Equal(t, calls, 1, "PodExec should be called once")
+		assert.Equal(t, pgadmin.Status.MajorVersion, 7)
+		assert.Equal(t, pgadmin.Status.ImageSHA, "newFakeSHA")
+	})
+}
+
+func TestReconcilePGAdminMajorVersion(t *testing.T) {
+	ctx := context.Background()
+	pod := corev1.Pod{}
+	pod.Namespace = "test-namespace"
+	pod.Name = "pgadmin-123-0"
+	reconciler := &PGAdminReconciler{}
+
+	podExecutor := func(
+		_ context.Context, stdin io.Reader, stdout, stderr io.Writer, command ...string,
+	) error {
+		return reconciler.PodExec(pod.Namespace, pod.Name, "pgadmin", stdin, stdout, stderr, command...)
+	}
+
+	t.Run("SuccessfulRetrieval", func(t *testing.T) {
+		reconciler.PodExec = func(
+			namespace, pod, container string,
+			stdin io.Reader, stdout, stderr io.Writer, command ...string,
+		) error {
+			assert.Equal(t, pod, "pgadmin-123-0")
+			assert.Equal(t, namespace, "test-namespace")
+			assert.Equal(t, container, naming.ContainerPGAdmin)
+
+			// Simulate a v7 version of pgAdmin by setting stdout to "7" for
+			// podexec call in reconcilePGAdminMajorVersion
+			stdout.Write([]byte("7"))
+			return nil
+		}
+
+		version, err := reconciler.reconcilePGAdminMajorVersion(ctx, podExecutor)
+		assert.NilError(t, err)
+		assert.Equal(t, version, 7)
+	})
+
+	t.Run("PodExecError", func(t *testing.T) {
+		reconciler.PodExec = func(
+			namespace, pod, container string,
+			stdin io.Reader, stdout, stderr io.Writer, command ...string,
+		) error {
+			return errors.New("PodExecError")
+		}
+
+		version, err := reconciler.reconcilePGAdminMajorVersion(ctx, podExecutor)
+		assert.Check(t, err != nil)
+		assert.Equal(t, version, 0)
+	})
+}
+
+func TestWritePGAdminUsers(t *testing.T) {
+	ctx := context.Background()
+	cc := setupKubernetes(t)
+	require.ParallelCapacity(t, 1)
+
+	recorder := events.NewRecorder(t, runtime.Scheme)
+	reconciler := &PGAdminReconciler{
+		Client:   cc,
+		Owner:    client.FieldOwner(t.Name()),
+		Recorder: recorder,
+	}
+
+	ns := setupNamespace(t, cc)
+	pgadmin := new(v1beta1.PGAdmin)
+	pgadmin.Name = "test-standalone-pgadmin"
+	pgadmin.Namespace = ns.Name
+	pgadmin.Spec.Users = []v1beta1.PGAdminUser{
```

Comment:
Looks like all be the first test "CreateOneUser" define this block themselves. Maybe move this into that test and leave it undefined globally :pick: